core/src/eval/mod.rs

```diff
@@ -396,40 +399,52 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
                 mut env,
             } = clos;
 
-            clos = match &*shared_term {
-                Term::Sealed(_, inner, lbl) => {
+            let has_cont_on_stack = self.stack.is_top_idx() || self.stack.is_top_cont();
+
+            clos = match_sharedterm!(match (shared_term) {
+                Term::Sealed(key, inner, label) => {
                     let stack_item = self.stack.peek_op_cont();
                     let closure = Closure {
-                        body: RichTerm {
-                            term: shared_term.clone(),
-                            pos,
-                        },
+                        body: RichTerm::new(Term::Sealed(key, inner.clone(), label.clone()), pos),
                         env: env.clone(),
```

Comment:
Because we need access to inner and the label as well. We need to construct a closure to match the signature of `update_at_indices` but then we need to destruct it and reconstruct it without the `sealed` inside. So we basically need to clone it for potential thunk updates.